keps/sig-apps/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -0,0 +1,453 @@
+    - DeleteOnStatefulSetDeletion
+      1. Create 2 pod statefulset, scale to 1 pod, confirm PVC still exists,
+      1. Create 2 pod statefulset, add data to PVs, scale to 1 pod, scale back to 2, confirm PV has data (PVC not deleted).
+      1. Create 2 pod statefulset, delete stateful set, confirm PVCs deleted
+      1. Create 2 pod statefulset, add data to PVs, manually delete one pod, confirm pod comes back and PV has data (PVC not deleted).
+      1. As above, but manually delete all pods in stateful set.
+      1. Create 2 pod statefulset, add data to PVs, manually delete one pod, immediately scale down to one pod, confirm PVC exists.
+      1. Create 2 pod statefulset, add data to PVs, manually delete one pod, immediately scale down to one pod, scale back to two pods, confirm PV has data.
+    - Retain: 
+      1. same tests as above, but PVCs not deleted in any case and confirm data intact on the PV.
+    - Pod restart tests:
+      1. Create statefulset, perform rolling update 
+1. Upgrade/Downgrade tests
+    1. Create statefulset in previous version and upgrade to the version 
+       supporting this feature. The PVCs should remain intact.
+    2. Downgrade to earlier version and check the PVCs with Retain
+       remain intact and the others with set policies before upgrade 
+       gets deleted based on if the references were already set.
+1. Feature disablement/enable test for alpha feature flag `statefulset-autodelete-pvcs`.
+
+
+### Graduation Criteria
+
+#### Alpha release
+- Complete adding the items in the 'Changes required' section.
+- Add unit, functional, upgrade and downgrade tests to automated k8s test.
+
+### Upgrade / Downgrade Strategy
+
+This features adds a new field to the StatefulSet. The default value for the new field
+maintains the existing behavior of StatefulSets.
+
+On a downgrade, the PersistentVolumeClaimReclaimPolicy field will be removed from any
+StatefulSets. If a scale-down or StatefulSet deletion is in process when the downgrade
+happens, any existing ownerRefs on PVCs will not be removed so that in most cases when the
+scale-down completes, unused PVCs will be deleted. However, there may be edge cases causing
+only some of the unused PVCs to be deleted. As unused PVCs remaining after a scaledown is
+the expected behavior of the downgraded clusters no further effort will be made to remove
+them.
+
+### Version Skew Strategy
+There is only kubecontroller manager changes involved, hence not applicable for
+version skew involving other components.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: StatefulSetAutoDeletePVC
+    - Components depending on the feature gate: kube-controller-manager
```

Comment:
technically, there will also be kube-apiserver involved in it
as you will need to handle new field there (i.e. dropDisableFields pattern, e.g.:
https://github.com/kubernetes/kubernetes/blob/97d40890d00acf721ecabb8c9a6fec3b3234b74b/pkg/api/pod/util.go#L442
)

Reply:
Added. I took the liberty of adding a breadcrumb to dropDisabledFields as well.